daemon/algod/api/client/restClient.go

```diff
@@ -434,6 +434,14 @@ func (client RestClient) PendingTransactionInformation(transactionID string) (re
 	return
 }
 
+// PendingTransactionInformationV2 gets information about a recently issued
+// transaction. See PendingTransactionInformation for more details.
+func (client RestClient) PendingTransactionInformationV2(transactionID string) (response generatedV2.PendingTransactionResponse, err error) {
+	transactionID = stripTransaction(transactionID)
+	err = client.get(&response, fmt.Sprintf("/v2/transactions/pending/%s", transactionID), nil)
+	return
+}
```

Comment:
This should probably be implemented with `versionAffinity`, but it doesn't have to be implemented with that on this PR.